test/specs/lib/factories-test.js

```diff
@@ -0,0 +1,268 @@
+import _ from 'lodash'
+import React, { isValidElement } from 'react'
+
+import { sandbox } from 'test/utils'
+import {
+  createShorthand,
+} from 'src/lib'
+
+// ----------------------------------------
+// Utils
+// ----------------------------------------
+
+/**
+ * Returns the result of a shorthand factory.
+ */
+const getShorthand = ({
+  Component = 'div',
+  mapValueToProps = val => ({}),
+  value,
+  defaultProps,
+}) => createShorthand(Component, mapValueToProps, value, defaultProps)
+
+// ----------------------------------------
+// Common tests
+// ----------------------------------------
+
+const itReturnsNull = (value) => {
+  it('returns null', () => {
+    expect(getShorthand({ value })).to.equal(null)
+  })
+}
+
+const itReturnsNullGivenDefaultProps = (value) => {
+  it('returns null given defaultProps object', () => {
+    expect(getShorthand({ value, defaultProps: { 'data-foo': 'foo' } })).to.equal(null)
+  })
+}
+const itReturnsNullGivenDefaultPropsFunc = (value) => {
+  it('returns null with defaultProps function', () => {
+    expect(getShorthand({ value, defaultProps: () => ({ 'data-foo': 'foo' }) })).to.equal(null)
+  })
+}
+
+const itReturnsAValidElement = (value) => {
+  it('returns a valid element', () => {
+    expect(isValidElement(getShorthand({ value }))).to.equal(true)
+  })
+}
+
+const itAppliesDefaultProps = (value) => {
+  it('applies defaultProps', () => {
+    const defaultProps = { some: 'defaults' }
+
+    shallow(getShorthand({ value, defaultProps }))
+      .props()
+      .should.deep.equal(defaultProps)
+  })
+}
+const itDoesNotIncludePropsFromMapValueToProps = (value) => {
+  it('does not include props from mapValueToProps', () => {
+    const props = { 'data-foo': 'foo' }
+    const wrapper = shallow(getShorthand({ value, mapValueToProps: () => props }))
+
+    _.each(props, (val, key) => {
+      wrapper.should.not.have.prop(key, val)
+    })
+  })
+}
+
+const itMergesClassNames = (extraClassName, classNameSource, shorthandConfig) => {
```

Comment:
Put `classNameSource` as first arg:
- matches `propsSource` being first in the functions below
- currently args 1 and 3 are used in the test and 2 is used to describe the test which is a little confusing.